### ISSUE TITLE
High Memory Usage and UI Freezing When Resizing Panels 😢

### ISSUE THREAD

**Issue by DevGlyph** [bvaughn/react-resizable-panels#315]
Hi there! 👋

First off, I want to express my admiration for the `react-resizable-panels project`. It's an incredibly useful component that's been a game-changer for the UI/UX of my application. The flexibility and ease of integration have genuinely impressed me, so kudos to you for developing such a fantastic tool!

However, I've encountered an issue that I hope you can help me with. While using the `PanelResizeHandle` to resize panels, I've noticed a significant increase in memory usage, which eventually leads to the UI becoming unresponsive. This issue occurs during the resizing action and seems to persist until the page is reloaded.

I've followed the standard usage guidelines as outlined in the documentation, encapsulating my content within `PanelGroup` and `Panel` components, and specifying a `PanelResizeHandle` for resizing operations. Here's a simplified version of my setup:

```jsx
<EditorProvider>
  <div className="app">
    <PanelGroup direction="horizontal">
      <Panel defaultSize={25}>
        <Sidebar />
      </Panel>
      <PanelResizeHandle
        style={{
          borderLeft: "1px solid #ccc",
        }}
      />
      <Panel defaultSize={75}>
        <div className="main-content">
          <TopBar />
          <EditorContainer />
        </div>
      </Panel>
    </PanelGroup>
  </div>
</EditorProvider>
```

I've also prepared a CodeSandbox example that reliably reproduces the issue, which you can find here: [Sandbox](https://codesandbox.io/p/github/DevGlyph/resiable-panels-problem/master?import=true)

I'm eager to find a solution or workaround for this problem, as `react-resizable-panels` plays a crucial role in my project. Any insights, suggestions, or fixes you could provide would be immensely appreciated. I'm looking forward to your expert advice and hoping there's a way to mitigate this performance issue.

Thank you so much for your time and for the incredible work on this component. Your dedication to open source and the developer community is truly inspiring.

**Comment by bvaughn** [comment]
Resizing a panel schedules a (React) update. If your component is doing something _really expensive_ during rendering, this could lead to a bottle neck. I don't know what `AffineEditorContainer` is but I assume it's slow to render.

I'd suggest you check your memoization logic (is it possible that you're re-rendering unnecessarily?) or, if that's not possible, try debouncing the updates to your slow component.

**Comment by bvaughn** [comment]
PS. Glad you're finding the library useful! :)

**Comment by fundon** [comment]
Pease check out https://github.com/toeverything/blocksuite/issues/6471#issuecomment-1993829592.

**Comment by bvaughn** [comment]
Oh, that's interesting. I'll file that as an upstream issue against the `stacking-order` library as well.

**Comment by bvaughn** [comment]
This fix has been published in react-resizable-panels@2.0.13

Thanks @fundon for lending a hand!

---

❤️ → ☕ [givebrian.coffee](http://givebrian.coffee/)
